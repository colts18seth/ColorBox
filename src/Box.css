.Box {
    margin: .5rem;
}