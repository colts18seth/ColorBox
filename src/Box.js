import React from 'react';
import './Box.css';

const Box = ({ width, height, color, remove }) => {

    const style = {
        width: Number(width),
        height: Number(height),
        backgroundColor: color
    }

    return (
        <li className="Box" style={style} data-testid="Box"  >
            <button onClick={remove}>X</button>
        </li>
    );
}

export default Box;