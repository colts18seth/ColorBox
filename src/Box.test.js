import React from 'react';
import { render } from '@testing-library/react';
import Box from './Box';

test('renders without crashing', () => {
    render(<Box />);
});

test('matches snapshot', () => {
    const { asFragment } = render(<Box />);
    expect(asFragment()).toMatchSnapshot();
});

it('renders "X" button', () => {
    const { getByText } = render(<Box />);
    const x = getByText('X');

    expect(x).toBeInTheDocument();
})