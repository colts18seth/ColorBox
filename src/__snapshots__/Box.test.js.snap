// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <li
    class="Box"
    data-testid="Box"
  >
    <button>
      X
    </button>
  </li>
</DocumentFragment>
`;
